docs/design-documents/README.md

```diff
@@ -0,0 +1,40 @@
+# Design Process for Mbed-OS
+This document defines the software design process for Mbed-OS. Anyone developing for Mbed-OS should use this process to submit the design to get it reviewed from the broader Mbed-OS developer community.
+
+## Design documents location and organization
+* It's suggested to write one design document per change, which could be a new feature, defect fixes, tools changes etc. Do not try to address multiple changes in the same document.
+* The design document will follow the format represented in [Design Document Template](https://github.com/ARMmbed/mbed-os/blob/master/docs/design-documents/design_template.md)
+
+* For each design submission, create a folder under *https://github.com/ARMmbed/mbed-os/docs/design-documents/{AREA OR COMPONENT}*. The area/component would be Drivers, Platform, Connectivity, Security, Peripherals etc. 
```

Comment:
Do we have areas/components listed anywhere? We got similar problem in labels here on github. I would like to document them and use them here/labels/elsewhere

Reply:
That's good idea, we can make a initial list. It can be expanded later as well.